src/app/services/tests/GameService.spec.ts

```diff
@@ -58,32 +58,34 @@ describe('GameService', () => {
         expect(service).toBeTruthy();
     });
     it('startObserving should delegate callback to partDao', () => {
-        const myCallback: (iPart: ICurrentPartId) => void = (iPart: ICurrentPartId) => {
-            expect(iPart.id).toBe('partId');
-        };
-        spyOn(partDao, 'getObsById').and.returnValue(of({ id: 'partId', doc: {
+        const part: IPart = {
             typeGame: 'Quarto',
             playerZero: 'creator',
             playerOne: 'joiner',
             turn: 2,
             listMoves: [MOVE_1, MOVE_2],
             result: MGPResult.UNACHIEVED.value,
-        } }));
+        };
+        const myCallback: (part: MGPOptional<IPart>) => void = (observedPart: MGPOptional<IPart>) => {
```

Comment:
const myCallback: (*observedPart*: MGPOptional<IPart>) => void = (observedPart: MGPOptional<IPart>)
For less confusion ?

Reply:
yes, chiant ce double-typage de fonctions